net/Socket.hpp

```diff
@@ -719,7 +719,8 @@ class SocketPoll
 
             std::lock_guard<std::mutex> lock(_mutex);
             _newSockets.emplace_back(std::move(newSocket));
-            wakeup();
+            if (!_newSockets.empty() || !_newCallbacks.empty())
```

Comment:
There is no need to check for `!_newSockets.empty()` here, since we just inserted an item in it, so this can never be false.

This means that this check is unnecessary, since the condition will always be true, and we indeed must always call `wakeup()` after inserting a new socket.

However, there is something better to do here! Which was done in `addCallback()` (before you removed it). So let me comment there, and you can apply the same idea here too.